src/java/org/apache/cassandra/db/streaming/CassandraStreamReader.java

```diff
@@ -187,8 +192,21 @@ protected SSTableTxnSingleStreamWriter createWriter(ColumnFamilyStore cfs, long
         StreamReceiver streamReceiver = session.getAggregator(tableId);
         Preconditions.checkState(streamReceiver instanceof CassandraStreamReceiver);
         ILifecycleTransaction txn = createTxn();
-        RangeAwareSSTableWriter writer = new RangeAwareSSTableWriter(cfs, estimatedKeys, repairedAt, pendingRepair, coordinatorLogOffsets, format, sstableLevel, totalSize, txn, getHeader(cfs.metadata()));
-        return new SSTableTxnSingleStreamWriter(txn, writer);
+        if (session.streamOperation() == StreamOperation.TRACKED_TRANSFER)
+        {
+            Preconditions.checkState(cfs.metadata().replicationType().isTracked());
+            File location = cfs.getDirectories().getPendingLocationForDisk(localDir, session.planId());
+            Descriptor desc = cfs.newSSTableDescriptor(location, format);
+            SSTableMultiWriter writer = SimpleSSTableMultiWriter.create(desc, estimatedKeys, ActiveRepairService.UNREPAIRED_SSTABLE, ActiveRepairService.NO_PENDING_REPAIR,
+                                                                                    coordinatorLogOffsets, cfs.metadata, null, sstableLevel, getHeader(cfs.metadata()),
+                                                                                    cfs.indexManager.listIndexGroups(), txn, cfs);
+            return new SSTableTxnSingleStreamWriter(txn, writer);
+        }
+        else
+        {
+            RangeAwareSSTableWriter writer = new RangeAwareSSTableWriter(cfs, estimatedKeys, repairedAt, pendingRepair, coordinatorLogOffsets, format, sstableLevel, totalSize, txn, getHeader(cfs.metadata()));
+            return new SSTableTxnSingleStreamWriter(txn, writer);
```

Comment:
nit: The `SSTableTxnSingleStreamWriter` is duplicated, although I don't know how we'd address that without something like simplifying the `SimpleSSTableMultiWriter` creation above. It's a little wacky how many time we pass `cfs` in the same `create()` argument list :D